sdk/python/kfp/components/structures/components.json_schema.outline.yaml

```diff
@@ -0,0 +1,77 @@
+<ComponentSpec>:
+    name: string
+    description: string
+    source: #SourceSpec
+        url: string
+    schemaVersion: string
+    inputs: #InputSpec[]
+        <InputSpec[]>:
+            name: string
+            type: TypeType
+            description: string
+            default: PrimitiveTypes
+            optional: boolean
+    outputs: #OutputSpec[]
+        <OutputSpec[]>:
+            name: string
+            type: TypeType
+            description: string
+    implementation: #ImplementationType
+        <oneOf>:
+            <ContainerImplementation>:
+                container: #ContainerSpec
+                    image: string
+                    command: #CommandlineArgumentType
+                        <oneOf>:
+                            <PrimitiveTypes>:
+                                <oneOf>: string, integer, number, boolean
+                            <InputValuePlaceholder>:
+                                inputValue: string
+                            <InputPathPlaceholder>:
+                                inputPath: string
+                            <OutputPathPlaceholder>:
+                                outputPath: string
+                            <ConcatPlaceholder>:
+                                concat: CommandlineArgumentType[]
+                            <IfPlaceholder>:
+                                if:
+                                    cond: IfConditionArgumentType
+                                    then: CommandlineArgumentType[]
+                                    else: CommandlineArgumentType[]
+                    args: CommandlineArgumentType
+                    env: Map[str, str]
+                    unconfigurableOutputPaths: Map[str, str]
+            <GraphImplementation>:
```

Comment:
I think we should restrict this to components for now, and leave graph components for later.

Reply:
Done